tests/test_cli.py

```diff
@@ -199,22 +198,17 @@ def test_data_and_data_file_mutually_exclusive(
             "copier",
             "copy",
             f"--data-file={local / 'data.yml'}",
-            '--data=question="also does not matter"',
+            f"--data=question={expected}",
             str(src),
             str(dst),
             "--quiet",
         ],
         exit=False,
     )
-    assert run_result[1] == 2  # cli failure
-    out, _ = capsys.readouterr()
-    assert any(
-        [
-            "Error: Given --data-file, the following are invalid ['--data']" in out,
-            "Error: Given --data, the following are invalid ['--data-file']" in out,
-        ]
-    )
-    assert not (dst / ".copier-answers.yml").exists()
+    assert run_result[1] == 0
+    answers = yaml.safe_load((dst / ".copier-answers.yml").read_text())
+    assert answers["_src_path"] == str(src)
+    assert answers["question"] == expected
```

Comment:
Could you please also assert that some other question that got answered only by yaml still keeps that value? You'll have to modify a bit more the copier.yml and input.yml files

Reply:
addressed in https://github.com/copier-org/copier/pull/1332/commits/69be4d11e47009e35112ba31067b73dc38010565